core/src/main/java/hudson/util/io/Archiver.java

```diff
@@ -36,11 +37,17 @@
  */
 public abstract class Archiver extends FileVisitor implements Closeable {
     protected int entriesWritten =0;
-
+    protected Consumer<String> consumer = (str)-> {};
     /**
      * Number of files/directories archived.
      */
     public int countEntries() {
         return entriesWritten;
     }
+
+    public Archiver withConsumer(Consumer<String> consumer){
```

Comment:
I suggest to use a name that makes us know the meaning of this consumer. A consumer to do what? do you know what I mean?

Reply:
is general purpose consumer, do not get wrong, but is a consumer to do whatever you want with the stashed files: list them, save them, print them...